gdx-backend-jtransc/src/com/jtransc/media/limelibgdx/LimeFileHandle.java

```diff
@@ -1,9 +1,223 @@
+			}
+		}
+		throw new GdxRuntimeException("Cannot write to an " + type + " file: " + file);
+	}
+
+	@Override
+	public Writer writer(boolean append, String charset) {
+		OutputStream output = write(append);
+
+		try {
+			return charset == null ? new OutputStreamWriter(output) : new OutputStreamWriter(output, charset);
+		} catch (UnsupportedEncodingException e) {
+			throw new GdxRuntimeException(e);
+		}
+	}
+
+	@Override
+	public boolean exists() {
+		if (type == FileType.Internal) {
+			return internalExist(file.getPath());
+		} else if (type == FileType.Classpath) {
+			return false;
+		}
+
+		return exist(file.getPath());
+	}
+
+	@Override
+	public void mkdirs() {
+		if (type == FileType.Classpath || type == FileType.Internal) {
+			throw new GdxRuntimeException("Cannot mkdirs with a " + type + " file: " + file);
+		}
+
+		mkdirs(file.getPath());
+	}
+
+	@Override
+	public FileHandle parent() {
+		String parent = file.getPath();
+
+		int index = parent.lastIndexOf('/');
+		if (index > 0) {
+			parent = parent.substring(0, index + 1);
+		}
+		if (parent.length() == 0 && type == FileType.Absolute) {
+			parent = "/";
+		}
+
+		return new LimeFileHandle(parent, type);
+	}
+
+	@Override
+	public boolean delete() {
+		if (type == FileType.Classpath || type == FileType.Internal) {
+			throw new GdxRuntimeException("Cannot delete a " + type + " file: " + file);
+		}
+		try {
+			delete(file.getPath());
+			return true;
+		} catch (Exception ignore) {
+		}
+		return false;
+	}
+
+	@Override
+	public boolean deleteDirectory() {
+		if (type == FileType.Classpath || type == FileType.Internal) {
+			throw new GdxRuntimeException("Cannot delete a " + type + " file: " + file);
+		}
+		try {
+			deleteDirectory(file.getPath());
+			return true;
+		} catch (Exception ignore) {
+			return false;
+		}
+	}
+
+	@Override
+	public String readString() {
+		InputStream is = null;
+
+		try {
+			is = read();
+			int length = is.available();
+			byte[] data = new byte[length];
+			is.read(data);
+			return new String(data);
+		} catch (Exception e) {
+			System.out.println("Cannot readString from file: " + file.getName() + " exception: " + e.toString());
+			return null;
+		} finally {
+			if (is != null) {
+				try {
+					is.close();
+				} catch (IOException ignore) {
+				}
+			}
+		}
+
+	}
+
+	@Override
+	public void writeString(String str, boolean append) {
+		OutputStream os = null;
+
+		try {
+			os = write(append);
+			byte[] data = str.getBytes();
+			os.write(data, 0, data.length);
+		} catch (Exception e) {
+			System.out.println("Cannot writeString to file: " + file.getName() + " exception: " + e.toString());
+		} finally {
+			if (os != null) {
+				try {
+					os.close();
+				} catch (IOException ignore) {
+				}
+			}
+		}
+	}
+
+	@Override
+	public long length() {
+		if (!exists()) {
+			return 0;
+		}
+		return fileSize(file.getPath());
 	}
+
+	@HaxeMethodBody("sys.FileSystem.deleteFile(p0._str);")
```

Comment:
instead of `p0._str` I would use `N.str(p0)`. That way I could change implementation later. (Not important, you can just ignore this)